### PR TITLE
DSPHLE: Remove individual byteswaps and replace with generic function.

### PR DESCRIPTION
**Pull request by phire**
This removes 7 GetPointers and quite a lot of duplicated code.

### CODE REVIEW

**Comment by phire** [comment]
Should be more or less ready to merge now.

**Comment by Mullin** [comment]
LGTM

File: Source/Core/Core/HW/Memmap.cpp [hunk 1]
```diff
@@ -347,4 +347,17 @@ bool IsRAMAddress(const u32 address, bool allow_locked_cache, bool allow_fake_vm
 	}
 }
 
+bool IsPhysical(const u32 address)
```
Comment (magumagu) [inline Source/Core/Core/HW/Memmap.cpp]: This looks suspicious; nothing outside of the CPU memory access functions should care about IsPhysical in this sense.  Everything else masks off the top bits.

**Comment by comex** [comment]
Please rebase this and remove the IsPhysical bit, as that seems to be independent of the rest of the PR.

File: Source/Core/Core/HW/DSPHLE/UCodes/AXVoice.h [hunk 2]
```diff
@@ -80,13 +80,11 @@ union AXBuffers
 // Read a PB from MRAM/ARAM
 bool ReadPB(u32 addr, PB_TYPE& pb)
 {
-	u16* dst = (u16*)&pb;
-	const u16* src = (const u16*)Memory::GetPointer(addr);
-	if (!src)
+	if(!Memory::IsPhysical(addr))
```
Comment (Tilka) [inline Source/Core/Core/HW/DSPHLE/UCodes/AXVoice.h]: If this is going to stay, please make it `if (`. Same below.

**Comment by shuffle2** [comment]
pls update @phire

**Comment by nbohr1more** [comment]
@dolphin-emu-bot rebuild

**Comment by degasus** [comment]
ping?

**Comment by delroth** [comment]
Last ping before closing this PR.

**Comment by phire** [comment]
Rebased and I remove the IsPhysical stuff.

File: Source/Core/Core/HW/Memmap.h [hunk 3]
```diff
@@ -92,4 +92,23 @@ void Write_U64(const u64 var, const u32 address);
 void Write_U32_Swap(const u32 var, const u32 address);
 void Write_U64_Swap(const u64 var, const u32 address);
 
+// Templated functions for byteswapped copies.
+template <typename T>
+void CopyFromEmuSwapped(T* data, u32 address, size_t size)
+{
+	const T* src = reinterpret_cast<T*>(GetPointer(address));
+
+	for (size_t i = 0; i < size / sizeof(T); i++)
```
Comment (degasus) [inline Source/Core/Core/HW/Memmap.h]: if (src != nullptr)

File: Source/Core/Core/HW/Memmap.h [hunk 4]
```diff
@@ -92,4 +92,23 @@ void Write_U64(const u64 var, const u32 address);
...
+		data[i] = Common::FromBigEndian(src[i]);
+}
+
+template <typename T>
+void CopyToEmuSwapped(u32 address, const T* data, size_t size)
+{
+	T* dest = reinterpret_cast<T*>(GetPointer(address));
+
+	for (size_t i = 0; i < size / sizeof(T); i++)
```
Comment (degasus) [inline Source/Core/Core/HW/Memmap.h]: if (dest != nullptr)

**Comment by nbohr1more** [comment]
Could've been a lucky run but this the first time I was able to start Super Mario Galaxy without a little audio stutter at the beginning. Even though HLE is broken in Rogue Leader, I also was able to play the last Death Star stage with no audio stutter here on the first run. Either I got incredibly lucky or this is a pretty nice improvement.

**Comment by degasus** [comment]
@nbohr1more There should be no improvement but code cleanup.

LGTM

**Comment by nbohr1more** [comment]
ping-a-roolian-ovski-ali-docious ?

**Comment by phire** [comment]
Oh right, I should merge this.

**Pull request merged**